source/particle/world.cc

```diff
@@ -1127,6 +1131,13 @@ namespace aspect
     void
     World<dim>::advance_timestep()
     {
+      // Generate more particles
+      if (generator->isTimeToGenerateParticles(this->get_timestep_number()))
+        {
+          generate_particles();
+          initialize_particles();
+        }
```

Comment:
I would like to find a different place for this piece of code (e.g. and update() function). One reason is that I am considering calling this function multiple time per timesteps to iterate over particle advection and Stokes solution, and in this case I do not want to generate particles in every iteration.